src/Amazon.ElasticBeanstalk.Tools/Commands/DeployEnvironmentCommand.cs

```diff
@@ -79,68 +68,70 @@ protected override void ParseCommandArguments(CommandOptions values)
-                {
-                    targetFramework = this.GetStringValueOrDefault(this.DeployEnvironmentOptions.TargetFramework, CommonDefinedCommandOptions.ARGUMENT_FRAMEWORK, true);
-                }
-            }
+                this.EnsureInProjectDirectory();
 
-                string application = this.GetStringValueOrDefault(this.DeployEnvironmentOptions.Application, EBDefinedCommandOptions.ARGUMENT_EB_APPLICATION, true);
-                string environment = this.GetStringValueOrDefault(this.DeployEnvironmentOptions.Environment, EBDefinedCommandOptions.ARGUMENT_EB_ENVIRONMENT, true);
-                string versionLabel = this.GetStringValueOrDefault(this.DeployEnvironmentOptions.VersionLabel, EBDefinedCommandOptions.ARGUMENT_EB_VERSION_LABEL, false) ?? DateTime.Now.Ticks.ToString();
+                var projectLocation = Utilities.DetermineProjectLocation(this.WorkingDirectory,
+                    this.GetStringValueOrDefault(this.ProjectLocation, CommonDefinedCommandOptions.ARGUMENT_PROJECT_LOCATION, false));
 
-            bool doesApplicationExist = await DoesApplicationExist(application);
-            bool doesEnvironmentExist = doesApplicationExist ? await DoesEnvironmentExist(application, environment) : false;
+                string configuration = this.GetStringValueOrDefault(this.DeployEnvironmentOptions.Configuration, CommonDefinedCommandOptions.ARGUMENT_CONFIGURATION, false) ?? "Release";
+                string targetFramework = this.GetStringValueOrDefault(this.DeployEnvironmentOptions.TargetFramework, CommonDefinedCommandOptions.ARGUMENT_FRAMEWORK, false);
+                string publishOptions = this.GetStringValueOrDefault(this.DeployEnvironmentOptions.PublishOptions, CommonDefinedCommandOptions.ARGUMENT_PUBLISH_OPTIONS, false);
 
-            if(!doesApplicationExist)
-            {
-                try
+                if (string.IsNullOrEmpty(targetFramework))
                 {
-                    this.Logger?.WriteLine("Creating new Elastic Beanstalk Application");
-                    await this.EBClient.CreateApplicationAsync(new CreateApplicationRequest
+                    targetFramework = Utilities.LookupTargetFrameworkFromProjectFile(projectLocation);
+                    if (string.IsNullOrEmpty(targetFramework))
                     {
-                        ApplicationName = application
-                    });
-                }
-                catch(Exception e)
-                {
-                    throw new ElasticBeanstalkExceptions("Error creating Elastic Beanstalk application: " + e.Message, ElasticBeanstalkExceptions.EBCode.FailedCreateApplication);
+                        targetFramework = this.GetStringValueOrDefault(this.DeployEnvironmentOptions.TargetFramework, CommonDefinedCommandOptions.ARGUMENT_FRAMEWORK, true);
+                    }
                 }
-            }
 
+                await CreateEBApplicationIfNotExist(application, doesApplicationExist);
```

Comment:
This line is repeated if the package does exist and if doesn't. Move this line to after the if/else and remove the duplicate from the else.